### PR TITLE
[FR]  Selection mark feature

### PR DESCRIPTION
**Pull request by mssfang**

### CODE REVIEW

File: sdk/formrecognizer/azure-ai-formrecognizer/src/main/java/com/azure/ai/formrecognizer/models/FormPage.java [hunk 1]
```diff
@@ -71,6 +76,32 @@ public FormPage(final float height, final float textAngle, final LengthUnit unit
...
+     * @param height the height of the image/PDF in pixels/inches, respectively.
+     * @param textAngle the general orientation of the text in clockwise direction.
+     * @param unit the unit used by the width, height and boundingBox properties.
+     * @param width the width of the image/PDF in pixels/inches, respectively.
+     * @param lines when includeFieldElements is set to true, a list of recognized text lines.
+     * @param tables list of data tables extracted from the page.
+     * @param pageNumber the 1-based page number in the input document.
+     * @param selectionMarks the selection marks in the input document.
+     */
+    public FormPage(final float height, final float textAngle, final LengthUnit unit,
```
Comment (samvaity) [inline sdk/formrecognizer/azure-ai-formrecognizer/src/main/java/com/azure/ai/formrecognizer/models/FormPage.java]: Don't need to have this.

Comment (mssfang) [reply sdk/formrecognizer/azure-ai-formrecognizer/src/main/java/com/azure/ai/formrecognizer/models/FormPage.java]: updated

File: sdk/formrecognizer/azure-ai-formrecognizer/src/samples/java/com/azure/ai/formrecognizer/RecognizeContentWithSelectionMarks.java [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+                    System.out.printf("Cell has text '%s', within bounding box %s.%n", formTableCell.getText(),
+                        boundingBoxStr);
+                });
+                System.out.println();
+            }
+            // Selection Mark information
+            final List<FormSelectionMark> selectionMarks = formPage.getSelectionMarks();
+            for (int j = 0; j < selectionMarks.size(); j++) {
+                final FormSelectionMark selectionMark = selectionMarks.get(i);
+                System.out.printf("Selection Mark, %s, has state = %s.%n", selectionMark.getText(), selectionMark.getState());
```
Comment (samvaity) [inline sdk/formrecognizer/azure-ai-formrecognizer/src/samples/java/com/azure/ai/formrecognizer/RecognizeContentWithSelectionMarks.java]: We should add one example to show the use of `asSelectionMark` too ?

Comment (samvaity) [reply sdk/formrecognizer/azure-ai-formrecognizer/src/samples/java/com/azure/ai/formrecognizer/RecognizeContentWithSelectionMarks.java]: selection.getState() -- should we this rather expose this as `selectionMark.isSelected()`

Comment (mssfang) [reply sdk/formrecognizer/azure-ai-formrecognizer/src/samples/java/com/azure/ai/formrecognizer/RecognizeContentWithSelectionMarks.java]: talked offline. Keep SelectionMarkState, service team will add "undetermined state" in the future.

And added the sample for asSelectionMark in receipt

File: sdk/formrecognizer/azure-ai-formrecognizer/src/samples/java/com/azure/ai/formrecognizer/RecognizeContentWithSelectionMarks.java [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+     *
+     * @param args Unused. Arguments to the program.
+     *
+     * @throws IOException Exception thrown when there is an error in reading all the bytes from the File.
+     */
+    public static void main(final String[] args) throws IOException {
+        // Instantiate a client that will be used to call the service.
+        FormRecognizerClient client = new FormRecognizerClientBuilder()
+                                          .credential(new AzureKeyCredential("{key}"))
+                                          .endpoint("https://{endpoint}.cognitiveservices.azure.com/")
```
Comment (samvaity) [inline sdk/formrecognizer/azure-ai-formrecognizer/src/samples/java/com/azure/ai/formrecognizer/RecognizeContentWithSelectionMarks.java:32]: nit: but is the formatting off here.

**Review by samvaity — APPROVED** [review]

File: sdk/formrecognizer/azure-ai-formrecognizer/src/samples/java/com/azure/ai/formrecognizer/RecognizeContentWithSelectionMarks.java (hunk 3 above)
Comment (mssfang) [reply sdk/formrecognizer/azure-ai-formrecognizer/src/samples/java/com/azure/ai/formrecognizer/RecognizeContentWithSelectionMarks.java:32]: noop

**Pull request merged**